src/App.tsx

```diff
@@ -8,52 +9,44 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
 import { Counter } from './features/counter/Counter';
+import { getUsers } from './api/users';
 
-export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+import { set as setUsers } from './features/users/usersSlice';
+import { loadPosts } from './features/post/postsSlice';
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+import { useAppDispatch, useAppSelector } from './app/hooks';
 
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { author } = useAppSelector(state => state.author);
+  const { loaded, hasError, posts } = useAppSelector(state => state.posts);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    getUsers()
+      .then(users => dispatch(setUsers(users)))
+      .catch(error => {
+        throw new Error(error);
```

Comment:
Instead of throwing an error, consider handling error state by dispatching an action to update error state in the Redux store.